content/post/twitter-hashtags-real-time.md

```diff
@@ -0,0 +1,560 @@
+    response = requests.get(query_url, auth=my_auth, stream=True)
+    return response
+
+
+# Create a TCP/IP socket
+sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+
+# Connect to Wallaroo
+wallaro_input_address = ('localhost', 8002)
+
+print 'connecting to Wallaroo on %s:%s' % wallaro_input_address
+sock.connect(wallaro_input_address)
+
+resp = get_tweets()
+send_tweets_to_wallaroo(resp,sock)
+```
+
+### 4. Create the Wallaroo Application
+
+Now we can build the Wallaroo application that identifies the trending hashtags on the real-time stream.
+
+The Wallaroo application logic is self-contained in `twitter_wallaroo_app.py`. We start by importing all the needed libraries.
+
+```python
+import struct
+import wallaroo
+import pandas as pd
+
+```
+
+### 5. Create The Decoder
+
+The [Decoder](https://docs.wallaroolabs.com/book/core-concepts/core-concepts.html) will translate the raw messages from the network connection and feed the resulting messages to the computations. Create a class called `Decoder` that implements the following three methods.
+
+* `header_length(self)`: This method returns a fixed integer that represents the number of bytes that hold the value of `payload_length`. In this case we return 5, which denotes that the value of `payload_length` is held in 5 bytes.
+
+* `payload_length(self, bs)`: This method takes a parameter `bs` which holds the number of bytes from `header_length` above; in our case 5 bytes of data. We then unpack these into an integer that denotes the length of the current message’s data payload to be read from the network stream.
+These are the same 5 bytes that are being sent by `send_tweets_to_wallaroo(http_resp, tcp_connection)` in the Twitter client. For example, if we receive `'00006'`, it is converted to integer value `6`, which tells wallaroo to read the next 6 bytes and give them to the `decode` method.
+
+* `decode(self, bs)`: this method reads the data payload sent through the network, the length of which was returned by `payload_length`. In this case, we convert the bytes to a UTF-8 string and pass that along to the computation.
+
+
+```python
+class Decoder(object):
+    def header_length(self):
+        return 5
+
+    def payload_length(self, bs):
+        return int(struct.unpack("!5s", bs)[0])
+
+    def decode(self, bs):
+        return bs.decode("utf-8")
+```
+
+### 6. Create The Computation
+
+Now let’s define the [computation](https://docs.wallaroolabs.com/book/core-concepts/core-concepts.html) class which will be used in the data processing logic. We will be applying this on the tweets received from the Decoder in order to extract all the Hashtags.
+
+Our computation is called `HashtagFinder`, and like all computations, it must implement the following two methods:
+
+* `name(self)`:  returns the name of the computation step.
+* `compute(self, input_data)`: takes a parameter `input_data`, containing the data coming from the previous step and implements the current step’s logic.
```

Comment:
Needs an explanation for why this is `compute` but the code below is `compute_multi`